crates/bevy_asset/src/lib.rs

```diff
@@ -105,26 +105,28 @@ impl Plugin for AssetPlugin {
             app.insert_resource(asset_server);
         }
 
-        app.add_stage_before(
-            bevy_app::CoreStage::PreUpdate,
-            AssetStage::LoadAssets,
-            SystemStage::parallel(),
+        app.register_type::<HandleId>();
+
+        app.configure_set(
+            AssetSet::LoadAssets
+                .no_default_set()
+                .before(CoreSet::PreUpdate),
         )
-        .add_stage_after(
-            bevy_app::CoreStage::PostUpdate,
-            AssetStage::AssetEvents,
-            SystemStage::parallel(),
+        .configure_set(
+            AssetSet::AssetEvents
+                .no_default_set()
+                .after(CoreSet::PostUpdate),
         )
-        .register_type::<HandleId>()
-        .add_system_to_stage(
-            bevy_app::CoreStage::PreUpdate,
-            asset_server::free_unused_assets_system,
-        );
+        .add_system(asset_server::free_unused_assets_system.in_set(CoreSet::PreUpdate));
 
         #[cfg(all(
             feature = "filesystem_watcher",
             all(not(target_arch = "wasm32"), not(target_os = "android"))
         ))]
-        app.add_system_to_stage(AssetStage::LoadAssets, io::filesystem_watcher_system);
+        app.add_system(
+            io::filesystem_watcher_system
+                .in_set(AssetSet::LoadAssets)
+                .in_set(CoreSet::Update),
```

Comment:
```suggestion
```
LoadAssets is configured in PreUpdate above. Seems like there's another loop somewhere else though as it still errors with roughly the same error when I try to run breakout.